code/game/objects/items/weapons/cigs.dm

```diff
@@ -160,6 +160,7 @@ LIGHTERS ARE IN LIGHTERS.DM
 	if(istype(glass))	//you can dip cigarettes into beakers
 		var/transfered = glass.reagents.trans_to(src, chem_volume)
 		if(transfered)	//if reagents were transfered, show the message
+			smoketime = initial(smoketime) * reagents.total_volume / reagents.maximum_volume
```

Comment:
Won't this decrease the smoketime? If `total_volume` is 30, and `maximum_volume` is 60, this will halve the time

Reply:
No. When you dip a cigarette or pipe into a beaker, it adds reagents. If you, after transferring reagents from the beaker to the cig/pipe after smoking a bit, have 30 / 60, it sets the current smoketime equal to half of the maximum, indicating there is half a cig left, since half the reagents remain.

Reply:
Let's say our cig has 300 seconds of smoketime
If we transfer 30u of reagents, we do the calculation `smoketime = 300 * 30 / 60 = 9000 / 60 = 150`. We halve our smoketime when we add half the capacity on an unsmoked cig
This should probably be `smoketime = smoketime + initial(smoketime) * reagents.total_volume * reagents.maximum_volume`, but this could also be used to make a cig last very long when adding reagents in 1u increments